TravelOptions/Scripts/TravelOptionsCastWhenHeld.cs

```diff
@@ -0,0 +1,116 @@
+using DaggerfallConnect.Save;
+using DaggerfallWorkshop;
+using DaggerfallWorkshop.Game;
+using DaggerfallWorkshop.Game.Entity;
+using DaggerfallWorkshop.Game.Formulas;
+using DaggerfallWorkshop.Game.Items;
+using DaggerfallWorkshop.Game.MagicAndEffects;
+using DaggerfallWorkshop.Game.MagicAndEffects.MagicEffects;
+
+namespace TravelOptions
+{
+    class TravelOptionsCastWhenHeld : CastWhenHeld
+    {
+        public override void SetProperties()
+        {
+            base.SetProperties();
+            properties.DisableReflectiveEnumeration = true;
+        }
+
+        const int normalMagicItemDegradeRate = 4;
+        const int restingMagicItemDegradeRate = 60;
+
+        public override PayloadCallbackResults? EnchantmentPayloadCallback(EnchantmentPayloadFlags context, EnchantmentParam? param = null, DaggerfallEntityBehaviour sourceEntity = null, DaggerfallEntityBehaviour targetEntity = null, DaggerfallUnityItem sourceItem = null, int sourceDamage = 0)
+        {
+            base.EnchantmentPayloadCallback(context, param, sourceEntity, targetEntity, sourceItem, sourceDamage);
+
+            // Validate
+            if ((context != EnchantmentPayloadFlags.Equipped &&
+                 context != EnchantmentPayloadFlags.MagicRound &&
+                 context != EnchantmentPayloadFlags.RerollEffect) ||
+                param == null || sourceEntity == null || sourceItem == null)
+                return null;
+
+            // Get caster effect manager
+            EntityEffectManager casterManager = sourceEntity.GetComponent<EntityEffectManager>();
+            if (!casterManager)
+                return null;
+
+            if (context == EnchantmentPayloadFlags.Equipped)
+            {
+                // Cast when held enchantment invokes a spell bundle that is permanent until item is removed
+                InstantiateSpellBundle(param.Value, sourceEntity, sourceItem, casterManager);
+            }
+            else if (context == EnchantmentPayloadFlags.MagicRound)
+            {
+                // Apply CastWhenHeld durability loss
+                ApplyDurabilityLoss(sourceItem, sourceEntity);
+            }
+            else if (context == EnchantmentPayloadFlags.RerollEffect)
+            {
+                // Recast spell bundle - previous instance has already been removed by EntityEffectManager prior to callback
+                InstantiateSpellBundle(param.Value, sourceEntity, sourceItem, casterManager, true);
+            }
+
+            return null;
+        }
+
+        void ApplyDurabilityLoss(DaggerfallUnityItem item, DaggerfallEntityBehaviour entity)
+        {
+            // TravelOptions change: don't decrease charge while tedious traveling, in order to replicate the behavior of fast travel
+            if (!GameManager.Instance.EntityEffectBroker.SyntheticTimeIncrease && !TravelOptionsMod.Instance.GetTravelControlUI().isShowing)
```

Comment:
This is the one line that wasn't copied from the DFU code. Only added the `&& !TravelOptionsMod.Instance.GetTravelControlUI().isShowing` part